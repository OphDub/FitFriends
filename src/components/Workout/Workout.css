.Workout {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: lightgrey;
  min-height: 75%;
  padding: 15px;
  margin: 0 10px 10px 10px;
  border-radius: 10px; }

.Workout-form {
  width: 100%;
  display: flex;
  flex-direction: column; }

.Workout-info {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 100%; }

.workout-name {
  margin-bottom: 15px; }

.workout-info-inputs {
  border: none;
  border-radius: 5px;
  padding: 10px;
  outline: none; }

.exercise-header {
  padding: 0;
  margin: 15px 0 0 0; }

.exercise-inputs {
  border: none;
  border-radius: 5px;
  padding: 5px;
  outline: none; }

.exercise-inputs::-webkit-inner-spin-button,
.exercise-inputs::-webkit-outer-spin-button {
  -webkit-appearance: none; }

.create-exercises {
  display: flex;
  justify-content: space-between; }

.exercise-name {
  width: 50%; }

.exercise-button,
.rendered-ex-btn {
  border: none;
  border-radius: 5px;
  padding: 5px;
  width: 30px; }

.exercise-button:hover,
.rendered-ex-btn:hover,
.Workout-post-btn:hover {
  background-color: black;
  color: white;
  cursor: pointer; }

.rendered-exercise {
  display: flex;
  flex-direction: row;
  justify-content: space-between; }

.rendered-ex-info {
  margin: 0px; }

.Workout-post-btn {
  border: none;
  border-radius: 5px;
  padding: 10px;
  margin-top: 15px; }

.Workout-post-btn h3 {
  margin: 0px; }
