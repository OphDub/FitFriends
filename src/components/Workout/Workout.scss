.Workout {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
}

.Workout-info {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.rendered-exercise {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}