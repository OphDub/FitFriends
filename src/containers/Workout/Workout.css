.Workout {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background-color: lightgrey;
  min-height: 75%;
  padding: 15px;
  margin: 0 10px 10px 10px;
  border-radius: 10px; }

.Workout-header {
  display: flex;
  flex-direction: column;
  align-content: flex-start; }

.Workout-header h3, p {
  margin: 5px 0; }

.Workout-form {
  width: 100%;
  display: flex;
  flex-direction: column; }

.Workout-info {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 100%; }

.workout-name {
  margin-bottom: 15px; }

.workout-info-inputs {
  border: none;
  border-radius: 5px;
  padding: 10px;
  outline: none;
  width: 95%; }

.workout-error {
  display: flex;
  justify-content: center; }

.exercise-header {
  padding: 0;
  margin: 15px 0 0 0; }

.exercise-inputs {
  border: none;
  border-radius: 5px;
  padding: 10px;
  outline: none; }

.exercise-inputs::-webkit-inner-spin-button,
.exercise-inputs::-webkit-outer-spin-button {
  -webkit-appearance: none; }

.create-exercises {
  display: flex;
  justify-content: space-between; }

.exercise-name {
  width: 50%; }

.exercise-button,
.rendered-ex-btn {
  border: none;
  border-radius: 5px;
  padding: 10px;
  width: 60px; }

.exercise-button:hover,
.rendered-ex-btn:hover,
.Workout-post-btn:hover {
  background-color: black;
  color: white;
  cursor: pointer; }

.rendered-exercise {
  display: flex;
  flex-direction: row;
  justify-content: space-between; }

.rendered-ex-info {
  margin: 0px; }

.Workout-post-btn {
  border: none;
  border-radius: 5px;
  padding: 10px;
  margin-top: 15px; }

.Workout-post-btn h3 {
  margin: 0px; }
