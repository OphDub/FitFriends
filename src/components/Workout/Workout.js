import React, { Component } from 'react';
import { postWorkout } from '../../actions/actionsIndex';
import { connect } from 'react-redux';
import './Workout.css';
// import FontAwesomeIcon from '@fortawesome/fontawesome';
// import faMinusSquare from '@fortawesome/fontawesome-free-solid/faMinusSquare';
// import faPlusSquare from '@fortawesome/fontawesome-free-solid/faPlusSquare';

class Workout extends Component {
  constructor () {
    super();
    this.state={
      workoutName: '',
      workoutDesc: '',
      reps: '',
      exercise: '',
      exercises: []
    }
  }

  handleChange = (e) => {
    const { value, name } = e.target;
    this.setState({[name]:value});
  }

  addExercise = (e) => {
    e.preventDefault();
    const { reps, exercise } = this.state;
    const id = Date.now()
    const newExercise = Object.assign({ id, reps, exercise});
    const exercises = [...this.state.exercises, newExercise]
    this.setState({
      exercises,
      reps: '',
      exercise: '',
    });
  }

  removeExercise = (e) => {
    e.preventDefault();
    const { id } = e.target

    const filtered = this.state.exercises.filter(exercise => exercise.id !== id)
    console.log(filtered);

    this.setState({ exercises: filtered });
  }

  renderExercises = () => {
    return this.state.exercises.map( (exercise,index) =>
      <li className="rendered-exercise"
        key={exercise.id}>
          <h6>{exercise.reps}</h6>
          <h6>{exercise.exercise}</h6>
        <button id={exercise.id}
          onClick={this.removeExercise}>
            -
        </button>
      </li>
    )
  }

  postWorkout = (e) => {
    e.preventDefault();
    const { workoutName, workoutDesc, exercises } = this.state;
    this.props.postWorkout({ workoutName, workoutDesc, exercises });

    this.setState({
      workoutName: '',
      workoutDesc: '',
      reps: '',
      exercise: '',
      exercises: [],
    })
  }

  render () {
    return(
      <section className="Workout">
        <form className="Workout-form">
          <div className="Workout-info">
            <input type="text"
              placeholder="Workout Name"
              name="workoutName"
              value={this.state.workoutName}
              onChange={this.handleChange}/>
            <input type="text"
              placeholder="Workout Description (Optional)"
              name="workoutDesc"
              value={this.state.workoutDesc}
              onChange={this.handleChange}/>
          </div>
          <div>
            <h4>Exercises</h4>
            <ul>
              {this.renderExercises()}
            </ul>
            <div>
              <input  type="number"
                placeholder="Reps"
                name="reps"
                value={this.state.reps}
                onChange={this.handleChange}/>
              <input  type="text"
                placeholder="Exercise"
                name="exercise"
                value={this.state.exercise}
                onChange={this.handleChange}/>
              <button onClick={this.addExercise}>
                +
              </button>
            </div>
          </div>
          <button className="Workout-post-btn"
            onClick={this.postWorkout}>
              Post Workout
          </button>
        </form>
      </section>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  postWorkout: (workout) => dispatch(postWorkout(workout))
})

export default connect(null, mapDispatchToProps)(Workout);