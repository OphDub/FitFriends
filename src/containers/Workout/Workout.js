import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Workout.css';
import { postWorkout } from '../../thunks/thunks';

export class Workout extends Component {
  constructor () {
    super();
    this.state={
      workoutName: '',
      workoutDesc: '',
      reps: '',
      exercise: '',
      exercises: [],
      errorMsg: ''
    };
  }

  handleChange = (e) => {
    const { value, name } = e.target;
    this.setState({[name]:value});
  }

  validateExercise = () => {
    const { exercises } = this.state;
    const message = 'Please add at least one exercise to your workout';
    let errorMsg;

    const validated = exercises.length <= 0 ?
      (errorMsg = message) && false : true;

    this.setState({ errorMsg });
    return validated;
  }

  validateWorkoutInfo = () => {
    const { workoutName, workoutDesc } = this.state;

    if (workoutName === '' || workoutDesc === '') {
      const errorMsg = 'Please give your workout a name and description';

      this.setState({ errorMsg });
      return false;
    }

    return true
  }

  addExercise = (e) => {
    e.preventDefault();
    const { reps, exercise } = this.state;
    const id = Date.now();
    const newExercise = Object.assign({ id, reps, exercise});
    const exercises = [...this.state.exercises, newExercise];

    this.setState({
      exercises,
      reps: '',
      exercise: ''
    });
  }

  removeExercise = (e) => {
    e.preventDefault();
    const { id } = e.target;
    const filtered = this.state.exercises.filter(exercise =>
      exercise.id !== parseInt(id));

    this.setState({ exercises: filtered });
  }

  postWorkout = (e) => {
    e.preventDefault();
    const { workoutName, workoutDesc, exercises } = this.state;

    if (!this.validateWorkoutInfo()) {
      return;
    }

    if (!this.validateExercise()) {
      return;
    }

    this.props.postWorkout({ workoutName, workoutDesc, exercises });

    this.setState({
      workoutName: '',
      workoutDesc: '',
      reps: '',
      exercise: '',
      exercises: []
    });
  }

  renderWorkoutForm = () => {
    return (
    <form className="Workout-form">
      <div className="Workout-info">
        <label className="workout-info-labels"htmlFor="workoutName">
          <p>Workout Name: </p>
          <input  type="text"
            className="workout-info-inputs workout-name"
            placeholder="Workout Name"
            name="workoutName"
            value={this.state.workoutName}
            onChange={this.handleChange}
          />
        </label>
        <label className="workout-info-labels"htmlFor="workoutDesc">
          <p>Description: </p>
          <input  type="text"
            className="workout-info-inputs workout-description"
            placeholder="Workout Description"
            name="workoutDesc"
            value={this.state.workoutDesc}
            onChange={this.handleChange}
          />
        </label>
      </div>
      <div>
        <h4 className="exercise-header">Exercises</h4>
        <p>Add or remove exercises for your workout:</p>
        <div className="create-exercises">
          <input  type="number"
            className="exercise-inputs exercise-number"
            placeholder="Reps"
            name="reps"
            value={this.state.reps}
            onChange={this.handleChange}
          />
          <input  type="text"
            className="exercise-inputs exercise-name"
            placeholder="Exercise"
            name="exercise"
            value={this.state.exercise}
            onChange={this.handleChange}
          />
          <button className="exercise-button"
            onClick={this.addExercise}>
            Add
          </button>
        </div>
        <ul className="rendered-exercises">
          {this.renderExercises()}
        </ul>
      </div>
      {this.renderError()}
      <button className="Workout-post-btn"
        onClick={this.postWorkout}>
        <h3>Post Workout</h3>
      </button>
    </form>
    )
  }

  renderExercises = () => {
    return this.state.exercises.map( (exercise) =>
      <li className="rendered-exercise"
        key={exercise.id}>
        <h5 className="rendered-ex-info">{exercise.reps}</h5>
        <h5 className="rendered-ex-info">{exercise.exercise}</h5>
        <button className="rendered-ex-btn"
          id={exercise.id}
          onClick={this.removeExercise}>
            Remove
        </button>
      </li>
    );
  }

  renderError = () => {
    return (
      <div className="workout-error">
        <h5 className="workout-error-msg">{this.state.errorMsg}</h5>
      </div>
    )
  }

  render () {
    return (
      <section className="Workout">
        <div className="Workout-header">
          <h3>Post Your Workout</h3>
          <p>Give it a name and any details for your teammates</p>
        </div>
        {this.renderWorkoutForm()}
      </section>
    );
  }
}

Workout.propTypes = {
  postWorkout: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  postWorkout: (workout) => dispatch(postWorkout(workout))
});

export default connect(null, mapDispatchToProps)(Workout);