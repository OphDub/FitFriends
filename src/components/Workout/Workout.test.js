import React from 'react';
import { shallow } from 'enzyme';
import { Workout, mapDispatchToProps } from './Workout';

describe('WORKOUT', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = shallow(<Workout />);
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should change state when handleChange is called', () => {
    const mockEvent = { target: { value: 'some workout', name: 'workoutName'}};
    const expected = 'some workout';

    renderedComponent.instance().handleChange(mockEvent);
    renderedComponent.update();

    expect(renderedComponent.state().workoutName).toEqual(expected);
  });

  it('should call postWorkout from props when postWorkout is called', () => {

  });

  it('should clear state when postWorkout is called', () => {

  });
})
