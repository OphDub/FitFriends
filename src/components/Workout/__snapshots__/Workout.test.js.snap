// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WORKOUT should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Workout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="Workout-form"
>
        <div
                className="Workout-info"
        >
                <input
                        name="workoutName"
                        onChange={[Function]}
                        placeholder="Workout Name"
                        type="text"
                        value=""
                />
                <input
                        name="workoutDesc"
                        onChange={[Function]}
                        placeholder="Workout Description (Optional)"
                        type="text"
                        value=""
                />
        </div>
        <div>
                <h4>
                        Exercises
                </h4>
                <ul
                        className="rendered-exercises"
                >
                        
                </ul>
                <div>
                        <input
                                name="reps"
                                onChange={[Function]}
                                placeholder="Reps"
                                type="number"
                                value=""
                        />
                        <input
                                name="exercise"
                                onChange={[Function]}
                                placeholder="Exercise"
                                type="text"
                                value=""
                        />
                        <button
                                onClick={[Function]}
                        >
                                +
                        </button>
                </div>
        </div>
        <button
                className="Workout-post-btn"
                onClick={[Function]}
        >
                Post Workout
        </button>
</form>,
      "className": "Workout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Workout-info"
>
            <input
                        name="workoutName"
                        onChange={[Function]}
                        placeholder="Workout Name"
                        type="text"
                        value=""
            />
            <input
                        name="workoutDesc"
                        onChange={[Function]}
                        placeholder="Workout Description (Optional)"
                        type="text"
                        value=""
            />
</div>,
          <div>
            <h4>
                        Exercises
            </h4>
            <ul
                        className="rendered-exercises"
            >
                        
            </ul>
            <div>
                        <input
                                    name="reps"
                                    onChange={[Function]}
                                    placeholder="Reps"
                                    type="number"
                                    value=""
                        />
                        <input
                                    name="exercise"
                                    onChange={[Function]}
                                    placeholder="Exercise"
                                    type="text"
                                    value=""
                        />
                        <button
                                    onClick={[Function]}
                        >
                                    +
                        </button>
            </div>
</div>,
          <button
            className="Workout-post-btn"
            onClick={[Function]}
>
            Post Workout
</button>,
        ],
        "className": "Workout-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="workoutName"
                onChange={[Function]}
                placeholder="Workout Name"
                type="text"
                value=""
/>,
              <input
                name="workoutDesc"
                onChange={[Function]}
                placeholder="Workout Description (Optional)"
                type="text"
                value=""
/>,
            ],
            "className": "Workout-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "workoutName",
                "onChange": [Function],
                "placeholder": "Workout Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "workoutDesc",
                "onChange": [Function],
                "placeholder": "Workout Description (Optional)",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Exercises
</h4>,
              <ul
                className="rendered-exercises"
>
                
</ul>,
              <div>
                <input
                                name="reps"
                                onChange={[Function]}
                                placeholder="Reps"
                                type="number"
                                value=""
                />
                <input
                                name="exercise"
                                onChange={[Function]}
                                placeholder="Exercise"
                                type="text"
                                value=""
                />
                <button
                                onClick={[Function]}
                >
                                +
                </button>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Exercises",
              },
              "ref": null,
              "rendered": "Exercises",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "rendered-exercises",
              },
              "ref": null,
              "rendered": Array [],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    name="reps"
                    onChange={[Function]}
                    placeholder="Reps"
                    type="number"
                    value=""
/>,
                  <input
                    name="exercise"
                    onChange={[Function]}
                    placeholder="Exercise"
                    type="text"
                    value=""
/>,
                  <button
                    onClick={[Function]}
>
                    +
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "reps",
                    "onChange": [Function],
                    "placeholder": "Reps",
                    "type": "number",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "exercise",
                    "onChange": [Function],
                    "placeholder": "Exercise",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "+",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "+",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Post Workout",
            "className": "Workout-post-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Post Workout",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="Workout-form"
>
          <div
                    className="Workout-info"
          >
                    <input
                              name="workoutName"
                              onChange={[Function]}
                              placeholder="Workout Name"
                              type="text"
                              value=""
                    />
                    <input
                              name="workoutDesc"
                              onChange={[Function]}
                              placeholder="Workout Description (Optional)"
                              type="text"
                              value=""
                    />
          </div>
          <div>
                    <h4>
                              Exercises
                    </h4>
                    <ul
                              className="rendered-exercises"
                    >
                              
                    </ul>
                    <div>
                              <input
                                        name="reps"
                                        onChange={[Function]}
                                        placeholder="Reps"
                                        type="number"
                                        value=""
                              />
                              <input
                                        name="exercise"
                                        onChange={[Function]}
                                        placeholder="Exercise"
                                        type="text"
                                        value=""
                              />
                              <button
                                        onClick={[Function]}
                              >
                                        +
                              </button>
                    </div>
          </div>
          <button
                    className="Workout-post-btn"
                    onClick={[Function]}
          >
                    Post Workout
          </button>
</form>,
        "className": "Workout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Workout-info"
>
              <input
                            name="workoutName"
                            onChange={[Function]}
                            placeholder="Workout Name"
                            type="text"
                            value=""
              />
              <input
                            name="workoutDesc"
                            onChange={[Function]}
                            placeholder="Workout Description (Optional)"
                            type="text"
                            value=""
              />
</div>,
            <div>
              <h4>
                            Exercises
              </h4>
              <ul
                            className="rendered-exercises"
              >
                            
              </ul>
              <div>
                            <input
                                          name="reps"
                                          onChange={[Function]}
                                          placeholder="Reps"
                                          type="number"
                                          value=""
                            />
                            <input
                                          name="exercise"
                                          onChange={[Function]}
                                          placeholder="Exercise"
                                          type="text"
                                          value=""
                            />
                            <button
                                          onClick={[Function]}
                            >
                                          +
                            </button>
              </div>
</div>,
            <button
              className="Workout-post-btn"
              onClick={[Function]}
>
              Post Workout
</button>,
          ],
          "className": "Workout-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  name="workoutName"
                  onChange={[Function]}
                  placeholder="Workout Name"
                  type="text"
                  value=""
/>,
                <input
                  name="workoutDesc"
                  onChange={[Function]}
                  placeholder="Workout Description (Optional)"
                  type="text"
                  value=""
/>,
              ],
              "className": "Workout-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "workoutName",
                  "onChange": [Function],
                  "placeholder": "Workout Name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "workoutDesc",
                  "onChange": [Function],
                  "placeholder": "Workout Description (Optional)",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Exercises
</h4>,
                <ul
                  className="rendered-exercises"
>
                  
</ul>,
                <div>
                  <input
                                    name="reps"
                                    onChange={[Function]}
                                    placeholder="Reps"
                                    type="number"
                                    value=""
                  />
                  <input
                                    name="exercise"
                                    onChange={[Function]}
                                    placeholder="Exercise"
                                    type="text"
                                    value=""
                  />
                  <button
                                    onClick={[Function]}
                  >
                                    +
                  </button>
</div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Exercises",
                },
                "ref": null,
                "rendered": "Exercises",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "rendered-exercises",
                },
                "ref": null,
                "rendered": Array [],
                "type": "ul",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      name="reps"
                      onChange={[Function]}
                      placeholder="Reps"
                      type="number"
                      value=""
/>,
                    <input
                      name="exercise"
                      onChange={[Function]}
                      placeholder="Exercise"
                      type="text"
                      value=""
/>,
                    <button
                      onClick={[Function]}
>
                      +
</button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "reps",
                      "onChange": [Function],
                      "placeholder": "Reps",
                      "type": "number",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "exercise",
                      "onChange": [Function],
                      "placeholder": "Exercise",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "+",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "+",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Post Workout",
              "className": "Workout-post-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Post Workout",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
